### DISCUSSION TITLE
Required fields are required when updating an item via GraphQL

### DISCUSSION THREAD

**Discussion by cosminepureanu** [directus/directus#5858] [Archive]
With GraphQL, I'm using the following mutation to update an item:

```graphql
mutation UpdateCoach($id: ID!, $data: update_coaches_input!) {
  update_coaches_item(id: $id, data: $data) {
    id
    first_name
    last_name
  }
}
```

The problem is that I always have to provide the `id` or other required fields in the `data` object. If I don't, then I get an error from Directus, something like:
```
"Variable \"$data\" got invalid value { first_name: \"Johnny\" }; Field \"status\" of required type \"String!\" was not provided."
```
The value of `data.id` doesn't really matter, so I always put 0 and it will update as expected.

If I try to update the item via the REST API, then I don't need to provide the required fields in the `data` object. To me, that works as expected. -- and I use it as a backup.

(The same thing happens when updating users)

It could be that I'm doing something wrong, but is this meant to be like this or is it a bug?

`Version: Directus 9.0.0-rc.68`

**Comment by rijkvanzanten** [comment]
Duplicate of https://github.com/directus/directus/issues/5723 👍🏻

**Reply by cosminepureanu** [reply]
Ok, I missed that one. Thanks for support!

**Reply by rijkvanzanten** [reply]
Graag gedaan 🙂
